lib/block-supports/layout.php

```diff
@@ -103,14 +103,26 @@ function gutenberg_get_layout_style( $selector, $layout, $has_block_gap_support
 			 */
 			if ( ! empty( $layout['justifyContent'] ) && array_key_exists( $layout['justifyContent'], $justify_content_options ) ) {
 				$style .= "justify-content: {$justify_content_options[ $layout['justifyContent'] ]};";
-				// --justification-setting allows children to inherit the value regardless or row or column direction.
-				$style .= "--justification-setting: {$justify_content_options[ $layout['justifyContent'] ]};";
+				if ( ! empty( $layout['setCascadingProperties'] ) && $layout['setCascadingProperties'] ) {
```

Comment:
My first reaction here is that I don't like this to be honest, it breaks the "layout" abstraction to solve a specific use-case.